packages/xchain-binance/src/client.ts

```diff
@@ -247,18 +222,15 @@ class Client implements BinanceClient, XChainClient {
    *
    * @throws {Error} Thrown if phrase has not been set before. A phrase is needed to create a wallet and to derive an address from it.
    */
-  getAddress = (): string => {
-    if (!this.address) {
-      const address = crypto.getAddressFromPrivateKey(this.getPrivateKey(), getPrefix(this.network))
-      if (!address) {
-        throw new Error(
-          'Address has to be set. Or set a phrase by calling `setPhrase` before to use an address of an imported key.',
-        )
-      }
-
-      this.address = address
+  getAddress = (index = 0): string => {
+    const address = crypto.getAddressFromPrivateKey(this.getPrivateKey(index), getPrefix(this.network))
+    if (!address) {
```

Comment:
do we need this check? it doesn't hurt but may not be neccessary..

Reply:
nice catch! updated with 4b76ae0